src/group_theory/perm/sign.lean

```diff
@@ -786,4 +787,192 @@ end
 
 end sign
 
+/--
+A permutation that swaps `i` with `i+1`.
+-/
+def adj_swap [add_monoid α] [has_one α] (i : α) : perm α := equiv.swap i (i + 1)
```

Comment:
Frustratingly I can't actually use this definition for `fin ℕ`

Reply:
```suggestion
variables {ι : Type*} [decidable_eq ι] [has_zero ι] [has_one ι] [has_add ι] -- castable to ℕ
/--
A permutation that swaps `i` with `i+1`.
-/
def adj_swap (i : ι) : perm ι := equiv.swap i (i + 1)

/-- The set of adjacent swaps needed to perform a move -/
def move_components (i j : ℕ) : list (perm ι) :=
(((list.range' j (i - j) ++ (list.range' i (j - i)).reverse).map nat.cast).map adj_swap)
```
What about that @eric-wieser?

Reply:
That's a good idea, thanks! I think I was shelving this PR until @Vierkantor wraps up or abandons their `has_enum` typeclass, or whatever it was; but I'll try that approach when I revive it!

Reply:
(or, feel free to adopt this branch yourself)